spec.html

```diff
@@ -37650,12 +37653,8 @@ <h1>Array.prototype.lastIndexOf ( _searchElement_ [ , _fromIndex_ ] )</h1>
           1. Let _O_ be ? ToObject(*this* value).
           1. Let _len_ be ? LengthOfArrayLike(_O_).
           1. If _len_ is 0, return *-1*<sub>𝔽</sub>.
-          1. If _fromIndex_ is present, let _n_ be ? ToIntegerOrInfinity(_fromIndex_); else let _n_ be _len_ - 1.
-          1. If _n_ is -&infin;, return *-1*<sub>𝔽</sub>.
-          1. If _n_ &ge; 0, then
-            1. Let _k_ be min(_n_, _len_ - 1).
-          1. Else,
-            1. Let _k_ be _len_ + _n_.
+          1. If _fromIndex_ is not present, let _k_ be _len_ - 1.
+          1. Else, let _k_ be min(? ToAbsoluteIndex(_fromIndex_, _len_), _len_ - 1).
```

Comment:
Any reason not to make this a positive test?

```suggestion
          1. If _fromIndex_ is present, let _k_ be min(? ToAbsoluteIndex(_fromIndex_, _len_), _len_ - 1).
          1. Else, let _k_ be _len_ - 1.
```

Reply:
Not particularly, but "_If \<input> is \<negative condition>, let \<position> be \<maximum>… else let \<position> be \<treat input numerically>_" is the overwhelmingly dominant pattern in such steps of the algorithms affected by this PR (although `{Array,%TypedArray%}.prototype.lastIndexOf` are unique in their condition distinguishing absence from undefined[^1]).

How strongly do you feel about this suggestion?

[^1]: …and `String.prototype.lastIndexOf` is unique in conflating non-numeric _end-position_ input with absent/undefined, as opposed to e.g. `slice` and `endsWith` and `subarray` :upside_down_face: